packages/bolt-connection/src/channel/node/node-channel.js

```diff
@@ -353,7 +355,27 @@ export default class NodeChannel {
       )
     })
 
-    this._conn.setTimeout(receiveTimeout)
+    this._receiveTimeout = receiveTimeout
+  }
+
+  /**
+   * Stops the receive timeout for the channel.
+   */
+  stopReceiveTimeout() {
+    if (this._receiveTimeout !== null && this._receiveTimeoutStarted) {
+      this._receiveTimeoutStarted = false
+      this._conn.setTimeout(0)
```

Comment:
do these `setTimeout` calls bode well with connection pooling/reuse?
could you end with a connection that has received a custom timeout for a previous op and remains with such a timeout for a new op (that would not normally time out)?

Reply:
Yes, It actuality solves this issue. The first implementation was setting the timeout when the HELLO SUCCESS was received and it just timeouts any time in the feature because of inactivity.

Now I am deferring the timeout set for whenever a request is sent and it is unset just after there aren't ongoing requests.

The implementation is not the most obvious since the `ChannelConnection` doesn't control the send/receive messages, it is done by the protocol (which is weird and I have to change it in the future).